File: common_java_lib/intent_uploader/src/com/whaley/biutil/intentuploader/BiLogUploaderFactory.java
package com.whaley.biutil.intentuploader;
import com.whaley.biutil.IBiLogUploader;

/**
 * Created by wanglei on 2016/7/28.
 */
public class BiLogUploaderFactory {
    public static IBiLogUploader CreateBiLogUploader()
    {
        return new BiIntentLogUploader();
    }
}

File: common_native_lib/biloguploaderfactory.cpp
/*
 * Yuchuan Sun
 * Version 1.0
 * Date 2016-8-5
 * BiLog common module that is used by native code
*/

#define LOG_TAG "BiLogUploaderFactory"
#include <utils/Log.h>

#include "biloguploaderfactory.h"
#include "biintentloguploader.h"

BiLogUploaderFactory::BiLogUploaderFactory()
{
    ALOGI("constructor");
}

BiLogUploaderFactory::~BiLogUploaderFactory()
{
    ALOGI("destructor");
}

android::sp<IBiLogUploader> BiLogUploaderFactory::CreateBiLogUploader()
{
    return new android::BiIntentLogUploader();
}

